File: WinPaletter/Languages/Lang_ReplaceText.vb
Public Class Lang_ReplaceText
    Private Sub Lang_ReplaceText_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Icon = Lang_JSON_Manage.Icon
        LoadLanguage
        ApplyStyle(Me)

    End Sub

    Public Function Replace(data As DataGridView, Column As Integer, FindWhat As String) As String
        Using dlg As New Lang_ReplaceText
            dlg.TextBox3.Text = FindWhat

            If dlg.ShowDialog() = DialogResult.OK Then

                Dim SearchText As String = dlg.TextBox3.Text
                Dim ReplaceBy As String = dlg.TextBox4.Text

                If String.IsNullOrWhiteSpace(SearchText) Then Return ReplaceBy

                For r = 0 To data.Rows.Count - 1
                    With data.Item(Column, r)
                        .Value = .Value.ToString.Replace(SearchText, ReplaceBy, Not dlg.CheckBox1.Checked, dlg.CheckBox2.Checked)
                    End With
                Next

                Return ReplaceBy

            Else
                Return dlg.TextBox4.Text
            End If
        End Using
    End Function

    Public Function Replace(Form As Form, FindWhat As String) As String
        Using dlg As New Lang_ReplaceText
            dlg.TextBox3.Text = FindWhat

            If dlg.ShowDialog() = DialogResult.OK Then

                Dim SearchText As String = dlg.TextBox3.Text
                Dim ReplaceBy As String = dlg.TextBox4.Text

                If String.IsNullOrWhiteSpace(SearchText) Then Return ReplaceBy

                For Each ctrl In Form.GetAllControls
                    If ctrl.Text IsNot Nothing AndAlso Not String.IsNullOrWhiteSpace(ctrl.Text) Then
                        ctrl.Text = ctrl.Text.Replace(SearchText, ReplaceBy, Not dlg.CheckBox1.Checked, dlg.CheckBox2.Checked)

                    ElseIf ctrl.Tag IsNot Nothing AndAlso Not String.IsNullOrWhiteSpace(ctrl.Tag.ToString) Then
                        ctrl.Tag = ctrl.Tag.ToString.Replace(SearchText, ReplaceBy, Not dlg.CheckBox1.Checked, dlg.CheckBox2.Checked)
                    End If
                Next

                Return ReplaceBy

            Else
                Return dlg.TextBox4.Text
            End If

        End Using
    End Function



    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        DialogResult = DialogResult.OK
        Close()
    End Sub

    Private Sub Button7_Click(sender As Object, e As EventArgs) Handles Button7.Click
        DialogResult = DialogResult.Cancel
        Close()
    End Sub
End Class

File: WinPaletter/Languages/Lang_ReplaceText.cs
using System;
using System.Windows.Forms;

namespace WinPaletter
{
    /// <summary>
    /// A form that allows the user to replace text in a DataGridView or a Form.
    /// </summary>
    public partial class Lang_ReplaceText
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Lang_ReplaceText"/> class.
        /// </summary>
        public Lang_ReplaceText()
        {
            InitializeComponent();
        }
        private void Lang_ReplaceText_Load(object sender, EventArgs e)
        {
            Icon = FormsExtensions.Icon<Lang_Editor>();
            this.LoadLanguage();
            ApplyStyle(this);

        }

        /// <summary>
        /// Replaces text in a DataGridView.
        /// </summary>
        /// <param name="data"></param>
        /// <param name="Column"></param>
        /// <param name="FindWhat"></param>
        /// <returns></returns>
        public string Replace(DataGridView data, int Column, string FindWhat)
        {
            // Hide the dialog
            using (Lang_ReplaceText dlg = new())
            {
                // Set the text to find
                dlg.TextBox3.Text = FindWhat;

                // If the user clicked OK, replace the text
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    // Get the search and replace text
                    string SearchText = dlg.TextBox3.Text;
                    string ReplaceBy = dlg.TextBox4.Text;

                    // If the search text is empty, return the replace text
                    if (string.IsNullOrWhiteSpace(SearchText)) return ReplaceBy;

                    // Replace the text in each cell in the specified column
                    for (int r = 0, loopTo = data.Rows.Count - 1; r <= loopTo; r++)
                    {
                        DataGridViewCell temp = data[Column, r];
                        temp.Value = temp.Value.ToString().Replace(SearchText, ReplaceBy, !dlg.CheckBox1.Checked, dlg.CheckBox2.Checked);
                    }

                    return ReplaceBy;
                }

                else
                {
                    return dlg.TextBox4.Text;
                }
            }
        }

        /// <summary>
        /// Replaces text in a Form.
        /// </summary>
        /// <param name="Form"></param>
        /// <param name="FindWhat"></param>
        /// <returns></returns>
        public string Replace(Form Form, string FindWhat)
        {
            // Hide the dialog
            using (Lang_ReplaceText dlg = new())
            {
                // Set the text to find
                dlg.TextBox3.Text = FindWhat;

                // If the user clicked OK, replace the text
                if (dlg.ShowDialog() == DialogResult.OK)
                {
                    // Get the search and replace text
                    string SearchText = dlg.TextBox3.Text;
                    string ReplaceBy = dlg.TextBox4.Text;

                    // If the search text is empty, return the replace text
                    if (string.IsNullOrWhiteSpace(SearchText)) return ReplaceBy;

                    // Replace the text in each control
                    foreach (Control ctrl in Form.GetAllControls())
                    {
                        // If the control has text, replace the text
                        if (ctrl.Text is not null && !string.IsNullOrWhiteSpace(ctrl.Text))
                        {
                            ctrl.Text = ctrl.Text.Replace(SearchText, ReplaceBy, !dlg.CheckBox1.Checked, dlg.CheckBox2.Checked);
                        }

                        // If the control has a tag, replace the tag
                        else if (ctrl.Tag is not null && !string.IsNullOrWhiteSpace(ctrl.Tag.ToString()))
                        {
                            ctrl.Tag = ctrl.Tag.ToString().Replace(SearchText, ReplaceBy, !dlg.CheckBox1.Checked, dlg.CheckBox2.Checked);
                        }
                    }

                    // Return the replace text
                    return ReplaceBy;
                }

                else
                {
                    // Return the replace text
                    return dlg.TextBox4.Text;
                }
            }
        }



        private void Button2_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.OK;
            Close();
        }

        private void Button7_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }
    }
}